Python/import.c

```diff
@@ -1226,10 +1256,22 @@ update_global_state_for_extension(PyThreadState *tstate,
         assert(def->m_base.m_copy == NULL);
     }
     else {
-        assert(def->m_base.m_init != NULL
-               || is_core_module(tstate->interp, name, path));
-        if (singlephase->m_dict == NULL) {
+        if (singlephase->m_init != NULL) {
+            assert(singlephase->m_dict == NULL);
+            assert(def->m_base.m_copy == NULL);
+            assert(def->m_size >= 0);
+            /* Remember pointer to module init function. */
+            // XXX Two modules should not share the same def->m_base.
+            //assert(def->m_base.m_init == NULL
+            //           || def->m_base.m_init == singlephase->m_init);
```

Comment:
Hm, I don't remember why this is here.
Let's have the tests find where this was needed? A debug-build assert probably is the best tool.
```suggestion
            // The following assert used to be skipped if
            //     def->m_base.m_init == singlephase->m_init
            // but two modules should not share the same def->m_base.
            assert(def->m_base.m_init == NULL);
```

Reply:
Yeah, that isn't very clear and the commented-out assert isn't helping matters.  I'll clarify the comment and drop the rest.

Reply:
done